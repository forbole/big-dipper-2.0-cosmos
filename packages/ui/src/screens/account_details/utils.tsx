import {
  AccountBalancesDocument,
  AccountCommissionDocument,
  AccountDelegationBalanceDocument,
  AccountDelegationRewardsDocument,
  AccountUnbondingBalanceDocument,
  AccountWithdrawalAddressDocument,
} from '@/graphql/general/account_details_documents';
import { toValidatorAddress } from '@/utils/prefix_convert';
import axios from 'axios';

import chainConfig from '@/chainConfig';

export const fetchCommission = async (address: string) => {
  const defaultReturnValue = {
    commission: {
      coins: [],
    },
  };
  try {
    const { data } = await axios.post(
      process.env.NEXT_PUBLIC_GRAPHQL_URL ||
        chainConfig.endpoints.graphql ||
        'http://localhost:3000/v1/graphql',
      {
        variables: {
          validatorAddress: toValidatorAddress(address),
        },
        query: AccountCommissionDocument,
      }
    );
    return data?.data ?? defaultReturnValue;
  } catch (error) {
    return defaultReturnValue;
  }
};

export const fetchAccountWithdrawalAddress = async (address: string) => {
  const defaultReturnValue = {
    withdrawalAddress: {
      address,
    },
  };
  try {
    const { data } = await axios.post(
      process.env.NEXT_PUBLIC_GRAPHQL_URL ||
        chainConfig.endpoints.graphql ||
        'http://localhost:3000/v1/graphql',
      {
        variables: {
          address,
        },
        query: AccountWithdrawalAddressDocument,
      }
    );
    return data?.data ?? defaultReturnValue;
  } catch (error) {
    return defaultReturnValue;
  }
};

export const fetchAvailableBalances = async (address: string) => {
  const defaultReturnValue = {
    accountBalances: {
      coins: [],
    },
  };
  try {
    const { data } = await axios.post(
      process.env.NEXT_PUBLIC_GRAPHQL_URL ||
        chainConfig.endpoints.graphql ||
        'http://localhost:3000/v1/graphql',
      {
        variables: {
          address,
        },
        query: AccountBalancesDocument,
      }
    );
    return data?.data ?? defaultReturnValue;
  } catch (error) {
    return defaultReturnValue;
  }
};

export const fetchDelegationBalance = async (address: string) => {
  const defaultReturnValue = {
    delegationBalance: {
      coins: [],
    },
  };
  try {
    const { data } = await axios.post(
      process.env.NEXT_PUBLIC_GRAPHQL_URL ||
        chainConfig.endpoints.graphql ||
        'http://localhost:3000/v1/graphql',
      {
        variables: {
          address,
        },
        query: AccountDelegationBalanceDocument,
      }
    );
    return data?.data ?? defaultReturnValue;
  } catch (error) {
    return defaultReturnValue;
  }
};

export const fetchUnbondingBalance = async (address: string) => {
  const defaultReturnValue = {
    unbondingBalance: {
      coins: [],
    },
  };
  try {
    const { data } = await axios.post(
      process.env.NEXT_PUBLIC_GRAPHQL_URL ||
        chainConfig.endpoints.graphql ||
        'http://localhost:3000/v1/graphql',
      {
        variables: {
          address,
        },
        query: AccountUnbondingBalanceDocument,
      }
    );
    return data?.data ?? defaultReturnValue;
  } catch (error) {
    return defaultReturnValue;
  }
};

export const fetchRewards = async (address: string) => {
  const defaultReturnValue = {
    delegationRewards: [],
  };
  try {
    const { data } = await axios.post(
      process.env.NEXT_PUBLIC_GRAPHQL_URL ||
        chainConfig.endpoints.graphql ||
        'http://localhost:3000/v1/graphql',
      {
        variables: {
          address,
        },
        query: AccountDelegationRewardsDocument,
      }
    );
    return data?.data ?? defaultReturnValue;
  } catch (error) {
    return defaultReturnValue;
  }
};
