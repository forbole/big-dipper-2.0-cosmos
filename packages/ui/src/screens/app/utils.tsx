import chainConfig from '@/chainConfig';

const { basePath, previewImage } = chainConfig();

export const OPEN_GRAPH_SEO = {
  type: 'website',
  site_name: 'Big Dipper',
  images: [
    {
      url: previewImage ?? '',
      width: 800,
      height: 600,
      alt: 'Preview Photo',
    },
  ],
};

export const TWITTER_SEO = {
  cardType: 'summary_large_image',
};

export const ADDITIONAL_LINK_TAGS_SEO = [
  {
    rel: 'apple-touch-icon',
    href: `${basePath}/icons/apple-touch-icon.png`,
    sizes: '180x180',
  },
  {
    rel: 'icon',
    type: 'image/png',
    href: `${basePath}/icons/favicon-32x32.png`,
    sizes: '32x32',
  },
  {
    rel: 'icon',
    type: 'image/png',
    href: `${basePath}/icons/favicon-16x16.png`,
    sizes: '16x16',
  },
  {
    rel: 'manifest',
    href: `${basePath}/icons/site.webmanifest`,
  },
  {
    rel: 'mask-icon',
    href: `${basePath}/icons/safari-pinned-tab.svg`,
    color: '#5bbad5',
  },
  {
    rel: 'shortcut icon',
    href: `${basePath}/icons/favicon.ico`,
  },
];

export const ADDITIONAL_META_TAGS = [
  {
    property: 'msapplication-TileColor',
    content: '#da532c',
  },
  {
    name: 'msapplication-config',
    content: `${basePath}/icons/browserconfig.xml`,
  },
];
