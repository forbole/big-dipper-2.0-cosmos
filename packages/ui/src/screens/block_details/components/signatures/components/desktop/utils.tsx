import AvatarName from '@/components/avatar_name';
import React, { ReactNode } from 'react';

export const columns: {
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'validator',
    width: 100,
  },
];

export const formatRows = (data: AvatarName[]) =>
  data.map((x): { [key: string]: ReactNode } => ({
    validator: <AvatarName address={x.address} imageUrl={x.imageUrl} name={x.name} />,
  }));
