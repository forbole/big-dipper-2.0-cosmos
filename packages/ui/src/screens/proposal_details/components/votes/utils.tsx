import type { VoteType } from '@/screens/proposal_details/components/votes/types';

export const getVoteKey = (vote: string) => {
  const votes = {
    VOTE_OPTION_YES: 'yes',
    VOTE_OPTION_NO: 'no',
    VOTE_OPTION_NO_WITH_VETO: 'veto',
    VOTE_OPTION_ABSTAIN: 'abstain',
    NOT_VOTED: 'notVoted',
  };

  return votes[vote as keyof typeof votes] || vote;
};

export const filterDataByTab = (props: { data: VoteType[]; notVoted: VoteType[]; tab: number }) => {
  if (props.tab === 5) {
    return props.notVoted;
  }

  return props.data.filter((x) => {
    if (props.tab === 1) {
      return x.vote === 'VOTE_OPTION_YES';
    }

    if (props.tab === 2) {
      return x.vote === 'VOTE_OPTION_NO';
    }

    if (props.tab === 3) {
      return x.vote === 'VOTE_OPTION_NO_WITH_VETO';
    }

    if (props.tab === 4) {
      return x.vote === 'VOTE_OPTION_ABSTAIN';
    }

    return true;
  });
};
