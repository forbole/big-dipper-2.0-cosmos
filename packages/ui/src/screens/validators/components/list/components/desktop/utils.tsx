import ConditionExplanation from '@/components/condition_explanation';
import InfoPopover from '@/components/info_popover';
import Typography from '@mui/material/Typography';
import type { TFunction } from '@/hooks/useAppTranslation';
import { ReactNode } from 'react';

export const fetchColumns = (
  t: TFunction
): {
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
  component?: ReactNode;
  sortKey?: string;
  sort?: boolean;
}[] => [
  {
    key: 'idx',
    width: 5,
  },
  {
    key: 'validator',
    sortKey: 'validator.name',
    width: 25,
    sort: true,
  },
  {
    key: 'votingPower',
    sortKey: 'votingPower',
    width: 30,
    sort: true,
  },
  {
    key: 'commission',
    sortKey: 'commission',
    align: 'right',
    width: 15,
    sort: true,
  },
  // {
  //   key: 'self',
  //   sortKey: 'selfPercent',
  //   align: 'right',
  //   width: 11,
  //   sort: true,
  // },
  // {
  //   key: 'delegators',
  //   sortKey: 'delegators',
  //   align: 'right',
  //   width: 11,
  //   sort: true,
  // },
  {
    key: 'condition',
    align: 'center',
    width: 15,
    component: (
      <Typography variant="h4" className="label popover">
        {t('condition')}
        <InfoPopover content={<ConditionExplanation />} />
      </Typography>
    ),
  },
  {
    key: 'status',
    width: 10,
  },
];
