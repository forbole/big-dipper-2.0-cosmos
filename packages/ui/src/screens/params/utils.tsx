import type { Distribution, Gov, Minting, Slashing, Staking } from '@/screens/params/types';
import { nanoToSeconds, secondsToDays } from '@/utils/time';
import type { TFunction } from '@/hooks/useAppTranslation';
import numeral from 'numeral';

const convertBySeconds = (seconds: number, t: TFunction) => {
  const SECONDS_IN_DAY = 86400;
  return seconds >= SECONDS_IN_DAY
    ? t('days', {
        day: secondsToDays(seconds),
      })
    : t('seconds', {
        second: seconds,
      });
};

export const formatStaking = (data: Staking, t: TFunction) => [
  {
    key: 'bondDenom',
    label: t('bondDenom'),
    detail: data.bondDenom,
  },
  {
    key: 'unbondingTime',
    label: t('unbondingTime'),
    detail: convertBySeconds(nanoToSeconds(data.unbondingTime), t),
  },
  {
    key: 'maxEntries',
    label: t('maxEntries'),
    detail: numeral(data.maxEntries).format('0,0'),
  },
  {
    key: 'historicalEntries',
    label: t('historicalEntries'),
    detail: numeral(data.historicalEntries).format('0,0'),
  },
  {
    key: 'maxValidators',
    label: t('maxValidators'),
    detail: numeral(data.maxValidators).format('0,0'),
  },
];

export const formatSlashing = (data: Slashing, t: TFunction) => [
  {
    key: 'downtimeJailDuration',
    label: t('downtimeJailDuration'),
    detail: t('seconds', {
      second: numeral(nanoToSeconds(data.downtimeJailDuration)).format('0,0'),
    }),
  },
  {
    key: 'minSignedPerWindow',
    label: t('minSignedPerWindow'),
    detail: `${numeral(data.minSignedPerWindow * 100).format('0.[00]')}%`,
  },
  {
    key: 'signedBlockWindow',
    label: t('signedBlockWindow'),
    detail: numeral(data.signedBlockWindow).format('0,0'),
  },
  {
    key: 'slashFractionDoubleSign',
    label: t('slashFractionDoubleSign'),
    detail: `${data.slashFractionDoubleSign * 100} / 100`,
  },
  {
    key: 'slashFractionDowntime',
    label: t('slashFractionDowntime'),
    detail: `${data.slashFractionDowntime * 10000} / ${numeral(10000).format('0,0')}`,
  },
];

export const formatMinting = (data: Minting, t: TFunction) => [
  {
    key: 'blocksPerYear',
    label: t('blocksPerYear'),
    detail: numeral(data.blocksPerYear).format('0,0'),
  },
  {
    key: 'goalBonded',
    label: t('goalBonded'),
    detail: `${numeral(data.goalBonded * 100).format('0.[00]')}%`,
  },
  {
    key: 'inflationMax',
    label: t('inflationMax'),
    detail: `${numeral(data.inflationMax * 100).format('0.[00]')}%`,
  },
  {
    key: 'inflationMin',
    label: t('inflationMin'),
    detail: `${numeral(data.inflationMin * 100).format('0.[00]')}%`,
  },
  {
    key: 'inflationRateChange',
    label: t('inflationRateChange'),
    detail: `${numeral(data.inflationRateChange * 100).format('0.[00]')}%`,
  },
  {
    key: 'mintDenom',
    label: t('mintDenom'),
    detail: data.mintDenom,
  },
];

export const formatDistribution = (data: Distribution, t: TFunction) => [
  {
    key: 'baseProposerReward',
    label: t('baseProposerReward'),
    detail: `${numeral(data.baseProposerReward * 100).format('0.[00]')}%`,
  },
  {
    key: 'bonusProposerReward',
    label: t('bonusProposerReward'),
    detail: `${numeral(data.bonusProposerReward * 100).format('0.[00]')}%`,
  },
  {
    key: 'communityTax',
    label: t('communityTax'),
    detail: `${numeral(data.communityTax * 100).format('0.[00]')}%`,
  },
  {
    key: 'withdrawAddressEnabled',
    label: t('withdrawAddressEnabled'),
    detail: `${data.withdrawAddressEnabled}`.toUpperCase(),
  },
];

export const formatGov = (data: Gov, t: TFunction) => [
  {
    key: 'minDeposit',
    label: t('minDeposit'),
    detail: `${data.minDeposit.value} ${data.minDeposit.displayDenom.toUpperCase()}`,
  },
  {
    key: 'maxDepositPeriod',
    label: t('maxDepositPeriod'),
    detail: convertBySeconds(nanoToSeconds(data.maxDepositPeriod), t),
  },
  {
    key: 'quorum',
    label: t('quorum'),
    detail: `${numeral(data.quorum * 100).format('0.[00]')}%`,
  },
  {
    key: 'threshold',
    label: t('threshold'),
    detail: `${numeral(data.threshold * 100).format('0.[00]')}%`,
  },
  {
    key: 'vetoThreshold',
    label: t('vetoThreshold'),
    detail: `${numeral(data.vetoThreshold * 100).format('0.[00]')}%`,
  },
  {
    key: 'votingPeriod',
    label: t('votingPeriod'),
    detail: convertBySeconds(nanoToSeconds(data.votingPeriod), t),
  },
];
