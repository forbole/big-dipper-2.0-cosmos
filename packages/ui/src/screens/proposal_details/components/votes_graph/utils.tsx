import { ThemeOptions } from '@mui/material/styles';
import Big from 'big.js';
import * as R from 'ramda';
import { formatNumber } from '@/utils/format_token';
import type { VotesType } from '@/screens/proposal_details/components/votes_graph/types';

type FormatGraphType = {
  data: VotesType;
  theme: ThemeOptions;
  total: Big;
};
export const formatGraphData = ({ data, theme, total }: FormatGraphType) => {
  const keys = R.keys(data);
  const color = {
    0: theme.palette?.custom?.charts.one,
    1: theme.palette?.custom?.charts.four,
    2: theme.palette?.custom?.charts.three,
    3: theme.palette?.custom?.charts.two,
  };

  const formattedData = keys.map((x, i) => {
    const selectedData = data[x];
    return {
      name: x as string,
      value: Big(selectedData.value).toNumber(),
      display: formatNumber(selectedData.value, selectedData.exponent),
      percentage: total.gt(0)
        ? `${Big(selectedData.value).div(total)?.times(100).toFixed(2)}%`
        : '0%',
      color: color[i as keyof typeof color],
    };
  });

  const notEmpty = formattedData.some((x) => x.value > 0);

  if (!notEmpty) {
    formattedData.push({
      name: 'empty',
      value: 2400,
      color: theme.palette?.custom?.charts.zero,
      percentage: '0%',
      display: '',
    });
  }

  return formattedData;
};
