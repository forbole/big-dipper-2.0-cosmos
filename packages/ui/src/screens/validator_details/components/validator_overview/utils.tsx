import { getValidatorStatus } from '@/utils/get_validator_status';

export const getStatusTheme = getValidatorStatus;

export const getCondition = (condition: number, status: number) => {
  let result = 'na';
  return result;
};
