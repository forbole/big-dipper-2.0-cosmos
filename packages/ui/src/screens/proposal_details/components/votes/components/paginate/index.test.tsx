import Paginate from '@/screens/proposal_details/components/votes/components/paginate';
import { MockTheme } from '@/tests/utils';
import renderer from 'react-test-renderer';

// ==================================
// mocks
// ==================================
jest.mock('@/components/pagination', () => (props: JSX.IntrinsicElements['div']) => (
  <div id="Pagination" {...props} />
));

// ==================================
// unit tests
// ==================================
describe('screen: TokenDetails/Paginate', () => {
  it('matches snapshot', () => {
    const component = renderer.create(
      <MockTheme>
        <Paginate
          {...{
            rowsPerPage: 10,
            page: 0,
            total: 100,
            handleChangePage: jest.fn(),
            handleChangeRowsPerPage: jest.fn(),
          }}
        />
      </MockTheme>
    );
    const tree = component?.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
