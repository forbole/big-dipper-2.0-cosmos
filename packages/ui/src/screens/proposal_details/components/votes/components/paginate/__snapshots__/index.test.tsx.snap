// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: TokenDetails/Paginate matches snapshot 1`] = `
<div
  className="makeStyles-root"
  handlePageChange={[MockFunction]}
  handleRowsPerPageChange={[MockFunction]}
  id="Pagination"
  page={0}
  rowsPerPage={10}
  rowsPerPageOptions={
    [
      10,
      25,
      50,
      100,
    ]
  }
  total={100}
/>
`;
